providers/dns/route53/route53.go

```diff
@@ -64,36 +85,38 @@ func (d customRetryer) RetryRules(r *request.Request) time.Duration {
 // public hosted zone via the FQDN.
 //
 // See also: https://github.com/aws/aws-sdk-go/wiki/configuring-sdk
-func NewDNSProvider() (*DNSProvider, error) {
-	hostedZoneID := os.Getenv("AWS_HOSTED_ZONE_ID")
+func NewDNSProvider(config *Config) (*DNSProvider, error) {
+	if config == nil {
+		config = NewDefaultConfig()
```

Comment:
I'd almost feel better if this did not perform a nil check here, and instead required the config to be passed. In `providers/dns/dns_providers.go`, it could just go `route53.NewDNSProvider(route53.NewDefaultConfig())`